packages/cli/src/cmds/validator/keymanager/keystoreCache.ts

```diff
@@ -0,0 +1,89 @@
+import fs from "node:fs";
+import bls from "@chainsafe/bls";
+import {Keystore} from "@chainsafe/bls-keystore";
+import {SignerLocal, SignerType} from "@lodestar/validator";
+import {fromHex, toHex} from "@lodestar/utils";
+import {PointFormat} from "@chainsafe/bls/types";
+import {lockFilepath, unlockFilepath} from "../../../util/lockfile.js";
+import {LocalKeystoreDefinition} from "./interface.js";
+
+export async function loadKeystoreCache(
+  cacheFilepath: string,
+  keystoreDefinitions: LocalKeystoreDefinition[]
+): Promise<SignerLocal[]> {
+  const keystores: Keystore[] = [];
+  const passwords: string[] = [];
+  for (const {keystorePath, password} of keystoreDefinitions) {
+    keystores.push(Keystore.parse(fs.readFileSync(keystorePath, "utf8")));
+    passwords.push(password);
+  }
+
+  if (keystores.length !== passwords.length) {
+    throw new Error(
+      `Number of keystores and passwords must be equal. keystores=${keystores.length}, passwords=${passwords.length}`
+    );
+  }
+
+  if (!fs.existsSync(cacheFilepath)) {
+    throw new Error(`Cache file ${cacheFilepath} does not exists.`);
+  }
+
+  lockFilepath(cacheFilepath);
+
+  const password = passwords.join("");
+  // We can't use Keystore.parse as it validates the `encrypted message` to be only 32 bytes.
+  const keystore = new Keystore(JSON.parse(fs.readFileSync(cacheFilepath, "utf8")));
+  const secretKeyConcatenatedBytes = await keystore.decrypt(password);
```

Comment:
What's the point of encrypting the keystore at all if we store the passwords in the same place? I assume the descrypting still takes a long time if the keystore is big but probably still better then decryption every single key separately as it was done before

It does not seem ideal to me that we store passwords in plain text at all which is currently done by Lodestar as we have a `keystore` and a `secrets` (with passwords) file. Those files are stored on the same device (and even the same folder) so it pointless in terms of security.

In my opinion, we should **never** store the password, only the first time the user imports a encrypted keystore the password should be interactively provided through the cli or keymanager API. After keystores are imported they should be stored unecrypted which makes subsequently loading them much faster and we never store passwords in plain text.

Reply:
That's what I figured, looks like all CLs implement it like this. We also quickly discussed this topic in the stand up, I will create a separate issue for this to get some ideas how we could improve the current security model but it might just be the case that a remoter signer/key manager is the only solution.

> So if we have 100 validators keys we can save upto 99% of time with the cache.

that's a great performance improvement, I was assuming it will be much faster but having constant time is really a big deal if you have a lot of validators